compiler/noirc_frontend/src/parser/parser.rs

```diff
@@ -530,11 +530,14 @@ impl<'a> Parser<'a> {
     }
 
     fn location_at_previous_token_end(&self) -> Location {
-        Location::new(self.span_at_previous_token_end(), self.previous_token_location.file)
+        let span_at_previous_token_end = Span::from(
+            self.previous_token_location.span.end()..self.previous_token_location.span.end(),
+        );
+        Location::new(span_at_previous_token_end, self.previous_token_location.file)
     }
 
-    fn span_at_previous_token_end(&self) -> Span {
-        Span::from(self.previous_token_location.span.end()..self.previous_token_location.span.end())
+    fn empty_ident_at_previous_token_end(&self) -> Ident {
+        Ident::new(String::new(), self.location_at_previous_token_end())
```

Comment:
I wonder if having the `Ident`'s string be empty could cause similar issues. Perhaps the default string could be something like `"{unknown}"` for clearer error messages?

Reply:
Done! I also pushed another commit to not produce a module AST if an identifier doesn't come after "mod" or "contract" because otherwise the error was a bit strange (it was already strange).